src/NCronJob/Scheduler/JobWorker.cs

```diff
@@ -191,4 +194,48 @@ public void ScheduleJob(JobDefinition job)
             run.NotifyStateChange(JobStateType.Scheduled);
         }
     }
+
+    public void RemoveJobByName(string jobName)
+    {
+        RemoveJobRunsByName(jobName);
+        registry.RemoveByName(jobName);
+    }
+
+    public void RemoveJobType(Type type)
+    {
+        var fullName = registry.FindJobDefinition(type)?.JobFullName;
+        if (fullName is null)
+        {
+            return;
+        }
+
+        registry.RemoveByType(type);
+        jobQueueManager.RemoveQueue(fullName);
+    }
+
+    public void RescheduleJobByName(JobDefinition jobDefinition)
+    {
+        ArgumentNullException.ThrowIfNull(jobDefinition);
+        ArgumentNullException.ThrowIfNull(jobDefinition.CustomName);
```

Comment:
If CustomName is a nullable property why are we expecting it to not be null here?

Reply:
The method signature tells you that - well more or less. `...ByName`. I can find a better signature for sure.

Reply:
I changed the name to increase verbosity